### PR TITLE
[sparse] add meta support for add operation (and copy)

### PR DESCRIPTION
**Pull request by aartbik**
This is a small step towards #117188
@pearu to review (this was split of #117907)

### CODE REVIEW

File: test/test_sparse.py [hunk 1]
```diff
@@ -4458,6 +4458,17 @@ def test_sum_meta(self, dtype, layout):
...
 
+    @all_sparse_layouts('layout', include_strided=False)
+    @parametrize("dtype", [torch.float64])
+    def test_add_meta(self, dtype, layout):
+        device = 'cpu'
+        index_dtype = torch.int64
+        for t in self.generate_simple_inputs(layout, device=device, dtype=dtype, index_dtype=index_dtype):
+            m = t.to(device='meta')
+            r = torch.add(m, m)
+            self.assertTrue(r.is_meta)
```
Comment (pearu) [inline test/test_sparse.py:4469]:
```suggestion
            self.assertEqual(r, m)
```
that will also check the shape and dtype properties of the tensors.

**Review by pearu — CHANGES_REQUESTED** [review]
I have a suggestion to testing meta tensor result but otherwise LGTM! Thanks, @aartbik!

File: test/test_sparse.py (hunk 1 above)
Comment (aartbik) [reply test/test_sparse.py:4469]: good idea, done!

**Review by pearu — APPROVED** [review]
LGTM! Thanks, @aartbik!

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**